langkit/support/langkit_support-lexical_env.adb

```diff
@@ -659,12 +659,24 @@ package body Langkit_Support.Lexical_Env is
 
          for C of All_Elements loop
             declare
-               Nodes : constant Internal_Map_Node_Vectors.Vector :=
+               I_Nodes : constant Internal_Map_Node_Vectors.Vector :=
                   Internal_Envs.Element (C).Native_Nodes;
+               F_Nodes : constant Internal_Map_Node_Maps.Map :=
+                  Internal_Envs.Element (C).Foreign_Nodes;
             begin
-               for I in reverse Nodes.First_Index .. Nodes.Last_Index loop
+
+               --  Append internal nodes
+               for I in reverse I_Nodes.First_Index .. I_Nodes.Last_Index loop
+                  Append_Result
+                    (I_Nodes.Get (I), Metadata, Current_Rebindings,
+                     From_Rebound);
+               end loop;
+
+               --  Append foreign nodes
+               for C in reverse F_Nodes.Iterate loop
```

Comment:
I suggest to remove the `reverse` here, to be consistent with the handling of foreign nodes in other lookups.

Reply:
Done! and added a testcase